pkg/jobs/registry.go

```diff
@@ -688,6 +691,98 @@ func (r *Registry) CreateIfNotExistAdoptableJobWithTxn(
 	return nil
 }
 
+const changefeedIdentityHashKey = "changefeed_identity_hash"
+
+// AddChangefeedIdentityHashToJobInfo adds a hash of a changefeed job's sink URI and
+// table IDs to the job info to help detect duplicate changefeed jobs.
+func (r *Registry) AddChangefeedIdentityHashToJobInfo(
+	ctx context.Context, details jobspb.ChangefeedDetails, infoStorage InfoStorage,
+) error {
+	changefeedHash, err := r.GetChangefeedIdentityHash(ctx, details)
+	if err != nil {
+		return err
+	}
+	return infoStorage.Write(ctx, changefeedIdentityHashKey, changefeedHash)
+}
+
+// GetChangefeedIdentityHash computes the hash of a changefeed job's identifying
+// attributes - sink URI and table IDs.
+func (r *Registry) GetChangefeedIdentityHash(
```

Comment:
nit: "Get" implies a field accessor. can you make the name more clearly indicate it's a calculation?